Neos.Neos/Classes/Fusion/Helper/BackendHelper.php

```diff
@@ -40,6 +41,39 @@ public function interfaceLanguage(): string
         return $this->userService->getInterfaceLanguage();
     }
 
+    public function isEditMode(ActionRequest $request): bool
+    {
+        return ($request->getControllerPackageKey() === 'Neos.Neos'
+            && $request->getControllerName() === "Frontend\Node"
+            && $request->getControllerActionName() === 'edit'
+        );
+    }
+
+    public function isPreviewMode(ActionRequest $request): bool
+    {
+        return ($request->getControllerPackageKey() === 'Neos.Neos'
+            && $request->getControllerName() === "Frontend\Node"
+            && $request->getControllerActionName() === 'preview'
+        );
+    }
+
+    public function renderingModeCacheIdentifier(ActionRequest $request): string
+    {
+        if (
+            $request->getControllerPackageKey() === 'Neos.Neos'
+            && $request->getControllerName() === "Frontend\Node"
+            && ($request->getControllerActionName() === 'edit' || $request->getControllerActionName() === 'preview')
+        ) {
+            $editPreviewModeArgument = $request->hasArgument('editPreviewMode') ? $request->getArgument('editPreviewMode') : null;
+            if (is_string($editPreviewModeArgument)) {
+                return $request->getControllerActionName() . ':' . $editPreviewModeArgument;
+            }
+            return $request->getControllerActionName();
+        } else {
+            return "show";
```

Comment:
```suggestion
            return null;
```

instead?

Reply:
:-/ that would mean to change the return type tp ?string ... i personally like cache identifier as string.

Reply:
I agree, nullable (return) types add complexity.
I'm just a bit unsure about the many magic strings in the code, but than again I think the actual value is not that important as long as it's unique, right?
nitpick: double quotes should be replaced though

Reply:
Added single quotes